### DISCUSSION TITLE
Is that server-side replace zlib with zlib-ng while the client-side still use original zlib implementation ok?

### DISCUSSION THREAD

**Discussion by lufeidefei** [zlib-ng/zlib-ng#1339] [Q&A]
We have a application that use the zlib to compress/decompress net data。And i want to use the zlib-ng in the server-side, while do not modify the codes of client-side (maybe users do not want to upgrade theirs app)。Is that zlib and zlib-ng totally compatible。

**Comment by mtl1979** [comment]
We are working on making zlib-ng 100% compatible when it comes to the compression format, but we can't guarantee it yet... We test with a lot of different files to see if zlib-ng can handle the files correctly. This includes even corrupt files and too small buffers.

**Reply by lufeidefei** [reply]
So, I think it should be ok? If it can not work，it should be a bug?

**Reply by mtl1979** [reply]
> So, I think it should be ok? If it can not work，it should be a bug?

Yes, it should be a bug if replacing zlib on server-side with zlib-ng will break the code on client-side.
